CMakeLists.txt

```diff
@@ -20,7 +20,20 @@ cmake_minimum_required(VERSION 3.15)
 cmake_policy(SET CMP0069 NEW)  # LTO support
 cmake_policy(SET CMP0091 NEW)  # MSVC runtime support
 
-project(astcenc VERSION 3.0.0)
+project(astcencoder VERSION 3.0.0)
+
+option(UNIVERSAL_BUILD "Build multi-architecture universal binary [only supported on MacOS]" OFF)
+
+if (CMAKE_OSX_ARCHITECTURES)
+    message(STATUS "Building universal binary for ${CMAKE_OSX_ARCHITECTURES}")
+    set(UNIVERSAL_BUILD ON)
+endif()
+
+if (${UNIVERSAL_BUILD})
+    if (NOT CMAKE_OSX_ARCHITECTURES)
+        message(FATAL_ERROR "UNIVERSAL_BUILD requires -DCMAKE_OSX_ARCHITECTURES=\"arch1;arch2;...\"")
+    endif()
+endif()
```

Comment:
Builds currently still require the ARCH setting on the configure line, which doesn't make sense. (Line 51)

Reply:
> I was expecting this is assuming native Arm64 build. Because even in -DARCH=aarch64 case it wasn't cross compiling for aarch64 AFAIK

Cross-compiling works fine - the Jenkins macOS machine is x86-64. For macOS "it just works", for Linux you need to set `CXX` to the cross compiler you want to use.

Reply:
> Agree it should be removed including the whole messages of backend - ON, backend - SKIPPED and backend - OFF what do you think?

The backend logging is useful, because CMake doesn't give useful errors if the user typos an option variable name, it just silently accepts it. Some type of feedback of what is actually getting built is required.

Reply:
ARCH is now removed completely from cmake as well as jenkin scripts.